Packs/GSuiteSecurityAlertCenter/Integrations/GSuiteSecurityAlertCenter/GSuiteSecurityAlertCenter.py

```diff
@@ -0,0 +1,719 @@
+                      'time interval parameter.',
+    'INVALID_PARAM_VALUE_ERROR': 'The given value for {0} parameter is invalid.'
+}
+
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%S.%fZ'
+SCOPES: Dict[str, List[str]] = {
+    'ALERT': ['https://www.googleapis.com/auth/apps.alerts']
+}
+ALERT_FEEDBACK_TYPES = ['alert_feedback_type_unspecified', 'not_useful', 'somewhat_useful', 'very_useful']
+
+
+def validate_date(first_fetch) -> str:
+    """
+    Check whether the date provided is valid.
+
+    :type first_fetch: string
+    :param first_fetch: date
+
+    :return: raise ValueError if validation fails, else return parsed date.
+    :rtype: str
+    """
+    date_time = arg_to_datetime(first_fetch, is_utc=True, settings={'STRICT_PARSING': True})
+    if date_time:
+        create_time = date_time.strftime(DATE_FORMAT)
+    else:
+        raise ValueError(MESSAGES['INVALID_PARAM_VALUE_ERROR'].format('first fetch time interval'))
+
+    current_time = datetime.utcnow()
+    date_time_obj = datetime.strptime(create_time, DATE_FORMAT)
+
+    if date_time_obj > current_time:
+        raise ValueError(MESSAGES['INVALID_PARAM_VALUE_ERROR'].format('first fetch time interval'))
+
+    return create_time
+
+
+def validate_params_for_fetch_incidents(params: Dict[str, Any], last_run: Dict) -> Tuple[Dict[str, Any], str]:
+    """
+    Validates parameters for fetch-incidents command.
+
+    :param params: parameters dictionary.
+    :param last_run: A dict with a key containing the latest incident created time we got
+        from last fetch
+
+    :return: A tuple containing two elements:
+            params (``Dict[str, int]``): fetch incident parameters.
+            last_fetch (``str``): updated last_fetch.
+    """
+    params = GSuiteClient.remove_empty_entities(params)
+
+    # get user provided fetch_first interval
+    first_fetch = params.get('first_fetch')
+    first_fetch = DEFAULT_FIRST_FETCH if not first_fetch else first_fetch
+
+    # Validate the first_fetch value
+    create_time = validate_date(first_fetch)
+
+    # get the last_fetch value
+    last_fetch = last_run['last_fetch'] if last_run.get('last_fetch') else create_time
+
+    # get the user provided max_fetch value
+    max_fetch = params.get('max_fetch', DEFAULT_MAX_FETCH)
+    max_fetch = GSuiteClient.validate_get_int(max_fetch, limit=1000, message=MESSAGES['MAX_INCIDENT_ERROR'])
+
+    next_page_token = last_run.get('next_page_token', '')
+
+    # set the filter with last_fetch value
+    alert_filter = f'createTime>"{last_fetch}"'
+
+    # get user provided filter value and validate it
+    advance_filter = params.get('filter', '').replace("'", '"')
+    if 'createTime' in advance_filter or 'create_time' in advance_filter:
+        raise ValueError(MESSAGES['INVALID_FILTER'])
+    if advance_filter != '':
+        alert_filter += f' AND {advance_filter}'
+
+    if 'type' not in alert_filter:
+        alert_types = params.get('alert_type', '').split(',')
```

Comment:
The additional Info of `filter` param contains following text: `Note: Takes higher precedence against filter parameters.`

We can still mention in `alert_type` param that it will be overridden if type is present in `filter` param. 

@reutshal Kindly confirm if we should mention it or not.

Reply:
Yes I think that will be best

Reply:
Will change as required.

Reply:
Did not require to change this as added multi_select option as part of this param